### ISSUE TITLE
Setting a backcolor for a textannotation causes the previous annotation to dissapear

### ISSUE THREAD

**Issue by dlukach** [oxyplot/oxyplot-avalonia#35]
As the title says, the backcolor property causes my text to disappear. When I remove background, the issue is gone.

Using Avalonia 0.10.6 on Windows 10 x64 and Raspbian OS
Using Oxplot.Core 2.0.0.0
Using Oxyplot Avalonia 0.10.0 modified to allow touch gesture to work

![image](https://user-images.githubusercontent.com/11093792/127909357-513a1d47-caa8-4b26-9366-1598bdb87316.png)

Here is my code to add a new point
  
```
_runCount++;
XAxis.Reset();
YAxis.Reset();
DataPoint point = new(_runCount, _engine.GetFinalDataPoint());
ChartSeries.Points.Add(point);
Chart.Annotations.Add(new TextAnnotation{TextPosition = point, 
Text = point.Y.ToString(_profile.YStringFormat),
Background = _annotationBackground, TextColor = _annotationForeground,
StrokeThickness = 1, Offset = new(0, -20), Layer = AnnotationLayer.AboveSeries});
XAxis.AbsoluteMinimum = XAxis.Minimum = 0;
XAxis.AbsoluteMaximum = XAxis.Maximum =_runCount + 1;
int interval = 1;
if (_runCount > 5)
    interval = (_runCount - _runCount % 5) / 5;
XAxis.MajorStep = interval;
YAxis.MaximumPadding = YAxis.MinimumPadding = 0.1;          
Chart.InvalidatePlot(true);
```

Here is my code to set up the graph, this is in single reading mode
```
 private void SetUpGraph()
        {
            if((int)_profile.DataCollectionType.Value == (int)CommonTypes.DataCollectionType.Continuous)
            {
                YAxis.Title = Language.Instance["StringLoad"] + " (" + GlobalVars.GlobalSettings.LoadUnit.Value + ")";
                YAxis.StringFormat = FrameHandler.LoadResolutionFormat;
                if ((int)_profile.XAxis.Value == (int)CommonTypes.XAxisType.Time) //Time
                {
                    XAxis.Title  = Language.Instance["StringTime"] + " (s)";
                    XAxis.StringFormat  = "0.00";
                }
                else //Distance
                {
                    XAxis.Title = Language.Instance["StringDistance"] + " (" + GlobalVars.GlobalSettings.DistanceUnit.Value + ")";
                    XAxis.StringFormat  = FrameHandler.DistanceResolutionFormat;
                }
            }
            else //Single Reading
            {
                XAxis.Title =  Language.Instance["StringReadingNumber"];
                YAxis.Title = _profile.ContinuousYAxisTitle; //TODO: Convert to language

                IsSingleReading = true;
                ChartSeries.MarkerSize = 6;
                ChartSeries.MarkerType = MarkerType.Diamond;
                ChartSeries.Points.Add(DataPoint.Undefined);
            }
            XAxis.AxisChanged += AxisChangedEvent;
            Chart.Axes.Add(XAxis);
            Chart.Axes.Add(YAxis);
            Chart.Series.Add(ChartSeries);
            Controller = new();
            Controller.UnbindTouchDown();
            Controller.BindTouchDown(PlotCommands.PanZoomByTouch);
```

**Comment by VisualMelon** [comment]
After discussing further on gitter, we think that the plot canvas is rendering things in the wrong order sometimes due to a bug in Avalonia (sort is unstable, so entries with the same z-index can become out of order). This bug is fixed in https://github.com/AvaloniaUI/Avalonia/pull/6271

@dlukach suggests and has tested a neat solution in the meantime setting the Z-index to elements that are created by `CanvasRenderContext.CreateAndAdd'1`

    canvas.Children.Add(element);
    element.ZIndex = canvas.Children.Count - 1;

This could become a PR; should check Avalonia release frequency and determine if the Z-index has any other implications.

**Comment by VisualMelon** [comment]
Should be fixed now that we target Avalonia 0.10.7.
